### DISCUSSION TITLE
How to create a adapter package?

### DISCUSSION THREAD

**Discussion by dhpagani** [webpack/webpack#15077] [Q&A] [answered]
Hi all! 

I'm trying to create a single package to be a 'bridge/adapter' for the bundling process. We have several projects that bundles in a similar way but manage versions and configurations are painful.

So the main ideia is to create a single package `my-builder`  to manage the dependencies and webpack configuration. 

I've tried using two strategies: The first is to create a CLI to invoke functions (that can have some parameters to customize some values) that runs webpack via CLI (spawning process). The second is to export functions (with those parameters) that invokes webpack via Node API. 

So, in `my-package`, I have `my-builder` as a devDependency, and tried to run the scripts but I'm always some kind of error:

```ModuleNotFoundError: Module not found: Error: Can't resolve 'thread-loader' in 'my-package path'
resolve 'thread-loader' in 'my-package path'
  Parsed request is a module
  using description file: my-package/package.json (relative path: .)
    resolve as module
      request is not managed by the pnpapi
        Your application tried to access thread-loader, but it isn't declared in your dependencies; this makes the require call ambiguous and unsound.
        Required package: thread-loader
        Required by: my-package path
```
The `webpack`, `thread-loader`, and all the others deps are in `my-builder's` package.json as a `dependency`.

I understood that message as I have to put `thread-loader` in `devDependencies` of my `my-package`, but It is declared in `my-package`'s dependencies and the script inside `my-package` is calling webpack api that calls thread-loader (and the entire tree). 

I know about https://github.com/arcanis/pnp-webpack-plugin but is for webpack@4 and also de docs of webpack@5 says that is natively compatible with pnp.

Am I thinking in wrong way to create `my-builder`? 
I'm missing something or the entire strategy really wont work ? 

Is there any way to have the same structure that, if i need to change something in the bundling process, I do it only once?

Thank you people! Happy new year!

**Comment by alexander-akait** [comment, marked as answer]
As you can see this error is not related to webpack, please use `require.resolve` in your configuration, i.e. `loader: require.resolve("thread-loader")` and check you have `thread-loader` in your dependencies
